File: leetcode/1624-largest-substring-between-two-equal-characters/maxLengthBetweenEqualCharacters_test.go
package leetcode

import (
	"fmt"
	"path/filepath"
	"testing"

	"github.com/edram/leetcode/util"
)

type MaxLengthBetweenEqualCharactersCase struct {
	Input string

	Output int
}

func TestMaxLengthBetweenEqualCharactersCase(t *testing.T) {
	var casePath, _ = filepath.Abs("./cases.json")
	payload, err := util.LoadCases[[]MaxLengthBetweenEqualCharactersCase](casePath)
	if err != nil {
		panic(err)
	}

	for i, testCase := range *payload {
		fmt.Printf("正在执行第 %d 个测试用例\n", i+1)
		fmt.Printf("nums: %v ==> %v \n", testCase.Input, testCase.Output)

		result := maxLengthBetweenEqualCharacters(testCase.Input)
		if result != testCase.Output {
			t.Errorf("期望 %v, 结果 %v", testCase.Output, result)
		}
	}
}

File: leetcode/1624-largest-substring-between-two-equal-characters/maxLengthBetweenEqualCharacters.test.ts
import cases from "./cases.json";
import maxLengthBetweenEqualCharacters from "./maxLengthBetweenEqualCharacters";

test("largest substring between two equal characters", () => {
  cases.forEach((testCase, index) => {
    console.log(`正在执行第 ${index + 1} 个测试用例\n`);
    const result = maxLengthBetweenEqualCharacters(testCase.input);
    console.log(
      `input: ${testCase.input} , output: ${testCase.output} ==> ${result} \n`
    );

    expect(result).toBe(testCase.output);
  });
});
